### DISCUSSION TITLE
[4.1] - Drop foreign key on migration - Error

### DISCUSSION THREAD

**Discussion by rhadamez** [adonisjs/core#1821] [Help] [answered]
**Hey!**

I'm trying to drop a table, but its primary is a foreign key in another table. So I have to drop the foreign in another table first...but I got error of syntax :/
I tried somethings in knex docs, but still got the error.

My life plot in image:

![toAdonisCommunity](https://user-images.githubusercontent.com/18365154/97713222-4f265800-1a9e-11eb-9eb7-53dd51cbb28c.png)

**Sorry about the image, but I think is more easily to understand the problem**

**Comment by rhadamez** [comment, marked as answer]
As I see, the command:
`alter table 'request_appointments' drop foreign key 'request_appointments_requested_appointment_status_id_foreign' as '0'`

The problem is with the "as '0'", it does not work...How could I take this out?

**Reply by Banashek** [reply]
In the knex docs, the second parameter is marked as `[foreignKeyName]`, which is intended to signify that it is optional, not an array of one (not the best choice haha).

I think if you change it to `table.dropForeign('requested_appointment_status_id', 'request_appointments_requested_appointment_status_id_foreign')` that it should work.

**Reply by rhadamez** [reply]
Oh man, so that was the problem...the array. Actually that array really doesn't maked any sense there. Thanks!
